### PR TITLE
[New] : Functionality to add ETF deductions to the payitems entity

### PR DESCRIPTION
**Pull request by Himashana2**

### CODE REVIEW

File: src/main/java/com/hris/HRIS/controller/EmployeePayItemController.java [hunk 1]
```diff
@@ -282,4 +283,130 @@ public ResponseEntity<ApiResponse> addNoPayHoursDeductions(@PathVariable String
         ApiResponse apiResponse = new ApiResponse(returnMsg);
         return ResponseEntity.ok(apiResponse);
     }
+
+    @PostMapping("{email}/EPFdeductions")
```
Comment (kavicastelo) [inline src/main/java/com/hris/HRIS/controller/EmployeePayItemController.java:287]: replace endpoint with this
`/{email}/EPFdeductions`

**Review by kavicastelo — CHANGES_REQUESTED** [review]
I found a couple of wrong APIs in your control and need to be replaced. 
you tried to send requests like this
`/api/v1/employee/payitems{email}/EPFdeductions`

These types of APIs will be complicated to handle further you need to separate request params from the request API
correct way
`/api/v1/employee/payitems/{email}/EPFdeductions`

Request changes need to be applied in next PR @Himashana2

**Review by kavicastelo — COMMENTED** [review]
Changes are approved. 

Explain to me why you added a new npm package called `array-flatten` to the client side.

**Comment by Himashana2** [comment]
> Changes are approved.
> 
> Explain to me why you added a new npm package called `array-flatten` to the client side.

When I'm trying to run the angular application, it shows an error because it says "array-flatten" is missing... an error something like that. So, I installed it.

**Comment by kavicastelo** [comment]
> > Changes are approved.
> > Explain to me why you added a new npm package called `array-flatten` to the client side.
> 
> When I'm trying to run the angular application, it shows an error because it says "array-flatten" is missing... an error something like that. So, I installed it.

*Interesting!*
Thank you for providing details

**Pull request merged**